lib/cea/cea708_window.js

```diff
@@ -0,0 +1,397 @@
+      row.push(null);
+    }
+    return row;
+  }
+
+  /**
+   * Sets the unicode value for a char at the current pen location.
+   * @param {!string} char
+   */
+  setCharacter(char) {
+    // Check if the pen is out of bounds.
+    if (!this.isPenInBounds_()) {
+      return;
+    }
+
+    const cea708Char = new shaka.cea.CeaUtils.StyledChar(
+        char, this.underline_, this.italics_,
+        this.backgroundColor_, this.textColor_);
+    this.memory_[this.row_][this.col_] = cea708Char;
+
+    // Increment column
+    this.col_ ++;
+  }
+
+  /**
+   * Erases a character from the buffer and moves the pen back.
+   */
+  backspace() {
+    if (!this.isPenInBounds_()) {
+      return;
+    }
+
+    // Check if a backspace can be done.
+    if (this.col_ <= 0 && this.row_ <= 0) {
+      return;
+    }
+
+    if (this.col_ <= 0) {
+      // Move pen back a row.
+      this.col_ = this.colCount_ - 1;
+      this.row_--;
+    } else {
+      // Move pen back a column.
+      this.col_--;
+    }
+
+    // Erase the character occupied at that position.
+    this.memory_[this.row_][this.col_] = null;
+  }
+
+  /**
+   * @private
+   */
+  isPenInBounds_() {
+    const inRowBounds = this.row_ < this.rowCount_ && this.row_ >= 0;
+    const inColBounds = this.col_ < this.colCount_ && this.col_ >= 0;
+    return inRowBounds && inColBounds;
+  }
+
+  /**
+   * @return {!boolean}
+   */
+  isVisible() {
+    return this.visible_;
+  }
+
+  /**
+   * Moves up <count> rows in the buffer.
+   * @param {!number} count
+   * @private
+   */
+  moveUpRows_(count) {
+    let dst = 0; // Row each row should be moved to.
+
+    // Move existing rows up by <count>.
+    for (let i = count; i < shaka.cea.Cea708Window.MAX_ROWS; i++, dst++) {
+      this.memory_[dst] = this.memory_[i];
+    }
+
+    // Create <count> new rows at the bottom.
+    for (let i = 0; i < count; i++, dst++) {
+      this.memory_[dst] = this.createNewRow_();
+    }
+  }
+
+  /**
+   * Handles CR. Increments row - if last row, "roll up" all rows by one.
+   */
+  carriageReturn() {
+    if (this.row_ + 1 >= this.rowCount_) {
+      this.moveUpRows_(1);
+      this.col_ = 0;
+      return;
+    }
+
+    this.row_++;
+    this.col_ = 0;
+  }
+
+  /**
+   * Handles HCR. Moves the pen to the beginning of the cur. row and clears it.
+   */
+  horizontalCarriageReturn() {
+    this.memory_[this.row_] = this.createNewRow_();
+    this.col_ = 0;
+  }
+
+  /**
+   * @param {!number} endTime
+   * @param {!number} serviceNumber Number of the service emitting this caption.
+   * @return {?shaka.cea.ICaptionDecoder.ClosedCaption}
+   */
+  forceEmit(endTime, serviceNumber) {
+    const stream = `svc${serviceNumber}`;
+    const topLevelCue = new shaka.text.Cue(
+        this.startTime_, endTime, /* payload= */ '');
+
+    if (this.justification_ === 0) {
+      // LEFT justified.
+      topLevelCue.textAlign = shaka.text.Cue.textAlign.LEFT;
+    } else if (this.justification_ === 1) {
```

Comment:
You should probably add a condition for center, too, instead of relying on the default setting.